jujugui/static/gui/src/app/components/deployment-flow/changes/changes.js

```diff
@@ -21,38 +21,49 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('deployment-changes', function() {
 
   juju.components.DeploymentChanges = React.createClass({
+    displayName: 'DeploymentChanges',
+
     propTypes: {
-      changes: React.PropTypes.object.isRequired,
-      generateAllChangeDescriptions: React.PropTypes.func.isRequired
+      generateAllChangeDescriptions: React.PropTypes.func.isRequired,
+      getCurrentChangeSet: React.PropTypes.func.isRequired
+    },
+
+    getInitialState: function() {
+      this.interval = null;
+      return {
+        changes: this.props.getCurrentChangeSet()
+      };
     },
 
-    /**
-      Generate the list of change items.
+    componentDidMount: function() {
+      // poll the changeset every second for updates as there is no way that
+      // we can be sure when a bundle import will be complete.
+      this.interval = setInterval(() => {
```

Comment:
I didn't like this fix much either :)  but dd can deploy either a bundle or an individual charm and I had found instances where it would take time to load the charm as well and at that point the changes were still incomplete. We need a better way to determine when a set of user actions have been completed but I don't think we're there yet.